src/components/Input/Input.tsx

```diff
@@ -0,0 +1,57 @@
+import { FC, useState } from "react";
+import { InputType, InputElementProps } from "../Input/type";
+import classes from './Input.module.scss';
+
+const Input: FC<InputElementProps> = ({type, placeHolder, name, required, id, options=[], value, label}: InputElementProps) => {
+  const [enteredValue, setValue] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const handleOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newValue = e.target.value;
+    setValue(newValue);
+    setErrorMessage(validation(type, enteredValue));
+  }
+  return (
+    <label className={classes.inputContainer}>
+      {<span className={classes.inputLabel}>{label}</span>}
+      <p className={classes.error}>{errorMessage}</p>
+      {type === InputType.SELECT?
+        (<select name={name} className={classes.inputItem} required={required}>
+          <option className={classes.inputItem} disabled selected value={value} >{placeHolder}</option>
+          {options.map((option) => <option key={`${id}${option.value}`} className={classes.inputItem} value={option.value}>{option.label}</option>)}
+        </select>):
+        (<input type={type} placeholder={placeHolder} name={name} onChange={handleOnChange} className={classes.inputItem} required={required}/>)}
+    </label>
+  )
+}
+export default Input;
+
+const validation = (type: InputType, value: string): JSX.Element | string => {
```

Comment:
Could you please put this function in a separate file in the same folder?

Reply:
that one is resolved, we decided to keep it like that because it's just one function.